### PR TITLE
ISPN-8357 Remove usages of deprecated Index.LOCAL

### PR DESCRIPTION
**Pull request by anistor**
https://issues.jboss.org/browse/ISPN-8357

### CODE REVIEW

File: core/src/main/java/org/infinispan/configuration/cache/AbstractStoreConfigurationBuilder.java [hunk 1]
```diff
@@ -189,7 +189,7 @@ private void validateStoreAttributes() {
             throw CONFIG.sharedStoreWithLocalCache();
 
          if (!preload && builder.indexing().enabled()
-               && builder.indexing().indexLocalOnly() && !getBuilder().template())
+               && builder.indexing().index() == Index.PRIMARY_OWNER && !getBuilder().template())
```
Comment (karesti) [inline core/src/main/java/org/infinispan/configuration/cache/AbstractStoreConfigurationBuilder.java]: call `isPrimaryOwner` method

Comment (anistor) [reply core/src/main/java/org/infinispan/configuration/cache/AbstractStoreConfigurationBuilder.java]: But I've just deprecated Index.isPrimaryOwner. All those isX methods in an enum that are equivalent to a simple == X are useless. Why not add that clutter for all enum members ? Why do we even have enums? :)   But there are exceptions where such methods make sense, like when they test that the enum member is part of a subset of the whole set. Index.isEnabled is such a legitimate case. Everything else is clutter and abuse.

Comment (karesti) [reply core/src/main/java/org/infinispan/configuration/cache/AbstractStoreConfigurationBuilder.java]: I didn't see that deprecation, cool then

**Comment by anistor** [comment]
I've just rebased this

**Comment by gustavonalle** [comment]
Merged thanks @anistor !

**Pull request merged**